### PR TITLE
Upgrade STS to v3.7.0

### PR DESCRIPTION
**Pull request by kongslund**

### CODE REVIEW

File: Casks/sts.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+  sha256 'bb7bc748541e552e39095fa10990923c3c17df28b140ef9559a6bbf1f4ed262a'
 
   module Utils
     def self.based_on_eclipse
-      '4.4.2'   # find eclipse version at http://spring.io/tools/sts/all
+      '4.5'   # find eclipse version at http://spring.io/tools/sts/all
     end
   end
 
-  url "http://dist.springsource.com/release/STS/#{version}.RELEASE/dist/e#{Utils.based_on_eclipse.gsub(/\.\d$/, '')}/spring-tool-suite-#{version}.RELEASE-e#{Utils.based_on_eclipse}-macosx-cocoa-x86_64.tar.gz"
```
Comment (vitorgalvao) [inline Casks/sts.rb]: You should leave the `gusb` part. I’m willing to bet it’ll need to be readded on a point release.

Comment (kongslund) [inline Casks/sts.rb]: Thank you for your input. I've updated the PR with a more generic solution.

Comment (vitorgalvao) [inline Casks/sts.rb]: That is a bunch of unnecessary additions. We don’t even like that we need to keep track of the Eclipse version, but do it out of need. Please make the change as requested.

File: Casks/sts.rb [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+      '4.5'   # find eclipse version at http://spring.io/tools/sts/all
+    end
+
+    def self.eclipse_version_major_minor
+      eclipse_version.split('.').slice(0, 2).join('.')
     end
   end
 
-  url "http://dist.springsource.com/release/STS/#{version}.RELEASE/dist/e#{Utils.based_on_eclipse.gsub(/\.\d$/, '')}/spring-tool-suite-#{version}.RELEASE-e#{Utils.based_on_eclipse}-macosx-cocoa-x86_64.tar.gz"
+  url "http://dist.springsource.com/release/STS/#{version}/dist/e#{Utils.eclipse_version_major_minor}/spring-tool-suite-#{version}-e#{Utils.eclipse_version}-macosx-cocoa-x86_64.tar.gz"
```
Comment (vitorgalvao) [inline Casks/sts.rb]: Please delete the added code, and make the change as requested. We have plans to simplify things like getting minor version number for all casks, and making such a change on a single cask makes that harder, not easier.

File: Casks/sts.rb (hunk 1 above)
Comment (kongslund) [inline Casks/sts.rb]: If I reintroduce the `gsub` part as you request then the url will be incorrect for this particular release.

**Pull request merged**